src/ur5e_arm.cpp

```diff
@@ -403,6 +408,47 @@ void UR5eArm::send_trajectory(const std::vector<vector6d_t>& p_p,
 
 // helper function to read a data packet and send a noop message
 void UR5eArm::read_and_noop() {
+    std::string status;
+    dashboard->commandSafetyStatus(status);
+    if (status.find(urcl::safetyStatusString(urcl::SafetyStatus::NORMAL)) == std::string::npos) {
+        estop.store(true);
+
+    } else {
+        // if in normal state
+        if (estop.load()) {
+            estop.store(false);
+            BOOST_LOG_TRIVIAL(info) << "recovering from e-stop" << std::endl;
+
+            driver->resetRTDEClient(path_offset + OUTPUT_RECIPE, path_offset + INPUT_RECIPE);
+
+            if (!dashboard->commandPowerOff()) {
+                BOOST_LOG_TRIVIAL(info) << "yo fail power down: " << status << std::endl;
+                throw std::runtime_error("couldn't power off arm");
+            }
+            if (!dashboard->commandPowerOn()) {
+                BOOST_LOG_TRIVIAL(info) << "yo fail power up: " << status << std::endl;
+                throw std::runtime_error("couldn't power on arm");
+            }
+            // Release the brakes
+            if (!dashboard->commandBrakeRelease()) {
+                BOOST_LOG_TRIVIAL(info) << "yo failed release: " << status << std::endl;
+                throw std::runtime_error("couldn't release the arm brakes");
+            }
+            BOOST_LOG_TRIVIAL(info) << "yo break released: " << status << std::endl;
+
+            bool send_prog = driver->sendRobotProgram();
+            // don't know what to do if this is false yet
+            BOOST_LOG_TRIVIAL(info) << "send robot program successful: " << send_prog << std::endl;
```

Comment:
all of these have ok bools to check, but we do not know what the behavior should be if any of these are false. we left the `throws` in the code until we have a plan